src/Native/System.Net.Security.Native/pal_gssapi.cpp

```diff
@@ -0,0 +1,257 @@
+    int statusType = isGssMechCode ? GSS_C_MECH_CODE : GSS_C_GSS_CODE;
+    *outBufferHandle = new gss_buffer_desc();
+    uint32_t majorStatus =
+        gss_display_status(minorStatus, statusValue, statusType, GSS_C_NO_OID, nullptr, *outBufferHandle);
+    NetSecurity_HandleError(majorStatus, outBufferHandle, statusLength);
+    return majorStatus;
+}
+
+extern "C" uint32_t
+NetSecurity_ImportUserName(uint32_t* minorStatus, char* inputName, uint32_t inputNameLen, gss_name_t* outputName)
+{
+    assert(outputName != nullptr);
+    assert(minorStatus != nullptr);
+    assert(inputName != nullptr);
+    assert(outputName != nullptr);
+    gss_buffer_desc inputNameBuffer{.length = inputNameLen, .value = inputName};
+    gss_OID nameType = const_cast<gss_OID>(GSS_C_NT_USER_NAME);
+    return gss_import_name(minorStatus, &inputNameBuffer, nameType, outputName);
+}
+
+extern "C" uint32_t
+NetSecurity_ImportPrincipalName(uint32_t* minorStatus, char* inputName, uint32_t inputNameLen, gss_name_t* outputName)
+{
+    assert(minorStatus != nullptr);
+    assert(inputNameLen >= 0);
+    assert(outputName != nullptr);
+    gss_buffer_desc inputNameBuffer{.length = inputNameLen, .value = inputName};
+    gss_OID nameType = const_cast<gss_OID>(GSS_KRB5_NT_PRINCIPAL_NAME);
+    return gss_import_name(minorStatus, &inputNameBuffer, nameType, outputName);
+}
+
+extern "C" uint32_t NetSecurity_InitSecContext(uint32_t* minorStatus,
+                                               gss_cred_id_t claimantCredHandle,
+                                               gss_ctx_id_t* contextHandle,
+                                               uint32_t isNtlm,
+                                               gss_name_t targetName,
+                                               uint32_t reqFlags,
+                                               uint8_t* inputBytes,
+                                               uint32_t inputLength,
+                                               gss_buffer_t* outBufferHandle,
+                                               uint32_t* outTokenLength,
+                                               uint32_t* retFlags)
+{
+    assert(isNtlm == 0 || isNtlm == 1);
+    assert(minorStatus != nullptr);
+    assert(contextHandle != nullptr);
+    assert(outBufferHandle != nullptr);
+    assert(outTokenLength != nullptr);
+    assert(retFlags != nullptr);
+#if HAVE_GSS_SPNEGO_MECHANISM
+    gss_OID desiredMech = isNtlm ? GSS_NTLM_MECHANISM : GSS_SPNEGO_MECHANISM;
+#else
+    assert(!isNtlm && "NTLM is not supported by MIT libgssapi_krb5");
+    char gss_mech_value[] = "\x2b\x06\x01\x05\x05\x02"; // Binary representation of SPNEGO Oid (RFC 4178)
+    gss_OID_desc gss_mech_spnego_OID_desc = {.length = 6, .value = static_cast<void*>(gss_mech_value)};
+    gss_OID desiredMech = &gss_mech_spnego_OID_desc;
+#endif
+
+    gss_buffer_desc inputToken{.length = UnsignedCast(inputLength), .value = inputBytes};
+    *outBufferHandle = new gss_buffer_desc();
```

Comment:
I don't see a delete for gss_buffer_desc anywhere...is this deleted somewhere else?